src/openssl/ssl/socket.cr

```diff
@@ -105,7 +105,7 @@ abstract class OpenSSL::SSL::Socket < IO
     count = slice.size
     return 0 if count == 0
 
-    LibSSL.ssl_read(@ssl, slice.pointer(count), count).tap do |bytes|
+    LibSSL.ssl_read(@ssl, slice.to_unsafe, count).tap do |bytes|
```

Comment:
I think you can just pass `slice` here, no need to invoke `to_unsafe` (this is called automatically for C bindings). Check this in other places too, please. Thank you!

Reply:
@asterite thank you for the quick review!
This is interesting to learn, I'll update the PR quickly. :)

Reply:
@asterite I tried to pass `slice` directly but it seems that calling `to_unsafe` is required in order to have a `Pointer` type.

Reply:
Ah... I guess the pointer type doesn't match then. It's fine like that.